sormas-backend/src/main/java/de/symeda/sormas/backend/facility/FacilityFacadeEjb.java

```diff
@@ -564,18 +564,44 @@ public long count(FacilityCriteria criteria) {
 	@Override
 	public void saveFacility(FacilityDto dto) throws ValidationRuntimeException {
 
-		if (dto.getType() == null
-			&& !FacilityDto.OTHER_FACILITY_UUID.equals(dto.getUuid())
-			&& !FacilityDto.NONE_FACILITY_UUID.equals(dto.getUuid())) {
-			throw new ValidationRuntimeException(I18nProperties.getValidationError(Validations.validFacilityType));
-		}
+		validateFacilityDto(dto);
 
 		Facility facility = facilityService.getByUuid(dto.getUuid());
 
 		if (facility == null && !getByNameAndType(dto.getName(), dto.getDistrict(), dto.getCommunity(), dto.getType(), true).isEmpty()) {
 			throw new ValidationRuntimeException(I18nProperties.getValidationError(Validations.importFacilityAlreadyExists));
 		}
 
+		facility = fillOrBuildEntity(dto, facility, true);
+		facilityService.ensurePersisted(facility);
+	}
+
+	@Override
+	public void mergeOrSaveFacility(FacilityDto dto) throws ValidationRuntimeException {
```

Comment:
I think it's probably better to extend the existing saveFacility function with a parameter (=> mergeWithExistingFacility) instead of having two separate methods that have a significant amount of shared code

Reply:
Done